### ISSUE TITLE
[bug] `no-missing-component-display-name` can not check `displayName` from `MemberExpression`

### ISSUE THREAD

**Issue by hyoban** [Rel1cx/eslint-react#174]
### Describe the bug

This should not report an error

```tsx
const someThing = {
  displayName: "someThing",
}
const Component = React.forwardRef(() => <div/>)
Component.displayName = someThing.displayName
```

Maybe add a check in `componentCollector`

```ts
const maybeRightValue = match(right)
    .with({ type: NodeType.Literal }, ({ value }) => O.some(value))
    .with({ type: NodeType.TemplateLiteral }, n => O.some(getStaticValue(n)?.value))
    .with({ type: NodeType.MemberExpression }, n => {
	// here
    })
    .with({ type: NodeType.Identifier }, n => {
      return F.pipe(
        findVariableByNameUpToGlobal(n.name, context.getScope()),
        O.flatMap(getVariableInit(0)),
        O.filter(isOneOf([NodeType.Literal, NodeType.TemplateLiteral])),
        O.map(getStaticValue),
        O.flatMapNullable(v => v?.value),
      );
    })
    .otherwise(O.none);
```

### Platform and versions

```Text
node: v18.18.2
@eslint-react: v0.8.10-beta.2
```
